### PR TITLE
Allow to read OpenStack config from the secret

### PR DESCRIPTION
**Pull request by Fedosin**
Currently OpenStack cloud provider reads user credentials from config
file, where data is stored in clear text. This approach is not recommended,
as it is a serious security issue.

This commit add an ability to read the config from secrets, if necessary.
To do so, two new parameters are added to the config: SecretNamespace and
SecretName. If they are specified, the provider will try to read config
from the secret.

/kind feature

```release-note
Allow to read OpenStack user credentials from a secret instead of a local config file.
```

### CODE REVIEW

File: pkg/cloudprovider/providers/openstack/openstack.go [hunk 1]
```diff
@@ -245,6 +254,62 @@ func configFromEnv() (cfg Config, ok bool) {
...
+		return fmt.Errorf("failed to get kubernetes client: %v", err)
+	}
+
+	secret, err := k8sClient.CoreV1().Secrets(secretNamespace).Get(secretName, metav1.GetOptions{})
+	if err != nil {
+		klog.Warningf("Cannot get secret %s in namespace %s. error: %q", secretName, secretNamespace, err)
+		return err
+	}
+
+	for credentialKey, credentialValue := range secret.Data {
```
Comment (flaper87) [inline pkg/cloudprovider/providers/openstack/openstack.go]: Can you just access `secret.Data["clouds.conf"] ?

I think you can simply access the key like this: `content, ok := secret.Data["clouds.conf]` which should already decode the value for you or return `ok == false` if the key doesn't exist.

Comment (Fedosin) [reply pkg/cloudprovider/providers/openstack/openstack.go]: done

**Comment by flaper87** [comment]
/approve

@dims @hogepodge u guys cool merging this? I think it's a great security enhancement for the cloud provider. We should prob get it in regardless of the frozen state of the cloud provider.

**Comment by flaper87** [comment]
oh, right. I can't approve patches here :smile:

**Comment by dims** [comment]
/approve
/lgtm

@Fedosin please ensure that the same functionality is available in the external cloud provider as well

**Comment by derekwaynecarr** [comment]
super happy the community was able to work together to get this change in.

**Pull request merged**